File: java/src/br/unifor/facilitador/infraestrutura/ConfigEspecialidadeMemoria.java
package br.unifor.facilitador.infraestrutura;

import br.unifor.facilitador.dominio.ConfiguracaoEspecialidade;
import br.unifor.facilitador.dominio.Especialidade;
import br.unifor.facilitador.dominio.TipoDocumento;
import br.unifor.facilitador.porta.ConfigEspecialidadeRepository;
import java.util.EnumMap;
import java.util.List;
import java.util.Map;

/**
 * Aqui fica visivel o ganho da refatoracao: cada especialidade e uma ENTRADA,
 * nao uma classe. Em producao isso vira linha de tabela no Postgres.
 */
public class ConfigEspecialidadeMemoria implements ConfigEspecialidadeRepository {

    private final Map<Especialidade, ConfiguracaoEspecialidade> tabela =
            new EnumMap<>(Especialidade.class);

    public ConfigEspecialidadeMemoria() {
        tabela.put(Especialidade.CARDIOLOGIA, new ConfiguracaoEspecialidade(
                Especialidade.CARDIOLOGIA,
                "Voce assiste um cardiologista. Resuma risco cardiovascular.",
                List.of(TipoDocumento.EXAME_PREVIO, TipoDocumento.LAUDO)));

        tabela.put(Especialidade.DERMATOLOGIA, new ConfiguracaoEspecialidade(
                Especialidade.DERMATOLOGIA,
                "Voce assiste um dermatologista. Descreva a lesao relatada.",
                List.of(TipoDocumento.EXAME_PREVIO)));
    }

    @Override
    public ConfiguracaoEspecialidade para(Especialidade e) {
        ConfiguracaoEspecialidade cfg = tabela.get(e);
        if (cfg == null) {
            throw new IllegalArgumentException("Especialidade sem configuracao: " + e);
        }
        return cfg;
    }
}

File: javascript/src/infraestrutura/ConfigEspecialidadeMemoria.js
import { ConfigEspecialidadeRepository } from '../porta/ConfigEspecialidadeRepository.js';
import { ConfiguracaoEspecialidade } from '../dominio/ConfiguracaoEspecialidade.js';
import { Especialidade } from '../dominio/Especialidade.js';
import { TipoDocumento } from '../dominio/TipoDocumento.js';

/**
 * Aqui fica visivel o ganho da refatoracao: cada especialidade e uma ENTRADA,
 * nao uma classe. Em producao isso vira linha de tabela no banco.
 */
export class ConfigEspecialidadeMemoria extends ConfigEspecialidadeRepository {
  #tabela = new Map([
    [Especialidade.CARDIOLOGIA, new ConfiguracaoEspecialidade(
      Especialidade.CARDIOLOGIA,
      'Voce assiste um cardiologista. Resuma risco cardiovascular.',
      [TipoDocumento.EXAME_PREVIO, TipoDocumento.LAUDO],
    )],
    [Especialidade.DERMATOLOGIA, new ConfiguracaoEspecialidade(
      Especialidade.DERMATOLOGIA,
      'Voce assiste um dermatologista. Descreva a lesao relatada.',
      [TipoDocumento.EXAME_PREVIO],
    )],
  ]);

  para(especialidade) {
    const cfg = this.#tabela.get(especialidade);
    if (!cfg) {
      throw new Error(`Especialidade sem configuracao: ${especialidade}`);
    }
    return cfg;
  }
}
